components/suggestion/suggestion_box.jsx

```diff
@@ -399,33 +353,82 @@ export default class SuggestionBox extends React.Component {
             }
         }
 
+        this.clear();
+
         this.refs.input.focus();
 
         for (const provider of this.props.providers) {
             if (provider.handleCompleteWord) {
                 provider.handleCompleteWord(term, matchedPretext);
             }
         }
+    }
+
+    selectNext = () => {
+        this.setSelectionByDelta(1);
+    }
+
+    selectPrevious = () => {
+        this.setSelectionByDelta(-1);
+    }
+
+    setSelectionByDelta = (delta) => {
+        let selectionIndex = this.state.terms.indexOf(this.state.selection);
+
+        if (selectionIndex === -1) {
+            // this should never happen since selection should always be in terms
+            throw new Error('selection is not in terms');
```

Comment:
Maybe just set the selectionIndex to 0 in this case to prevent React from freaking out when an error is thrown

Reply:
Just copy code, but I will change it.